### PR TITLE
feat(groupchildren utility): adds a groupChildren uility to core

### PR DESCRIPTION
**Pull request by jameswburke**
This utility returns an object where the components are mapped by the group prop/config value.

## Issue(s): Relates to or closes...
https://alleyinteractive.atlassian.net/browse/IRV-726

## Summary: This pull request will...
Add a utility which maps an array of children into an object using each child's `group` prop.

## Example
```
const components = [
  {
    name: "irving/example-a",
    config: {
      "group": "first",
    },
  },
  {
    name: "irving/example-b",
    config: {
      "group": "first",
    },
  },
  {
    name: "irving/example-c",
    config: {
      "group": "second",
    },
  },
  {
    name: "irving/example-d",
  },
];

const groups = groupChildren(components);

console.log(groups);
```

Output:
```
{
  first: [
    {
      name: "irving/example-a",
      config: {
        "group": "first",
      },
    },
    {
      name: "irving/example-b",
      config: {
        "group": "first",
      },
    },
  ],
  second: [
    {
      name: "irving/example-c",
      config: {
        "group": "second",
      },
    },
  ],
  rest: [
    {
      name: "irving/example-d",
    }
  ]
}
```

## Tests: I know this code works because...
This utility has been battle tested on a few projects, and has unit tests.

### CODE REVIEW

File: packages/core/utils/children.test.js [hunk 1]
```diff
@@ -80,3 +81,74 @@ it('should filter an array of react elements by api component name', () => {
     },
   ]);
 });
+
+const testGroupChildrenElement = {
```
Comment (joemcgill) [inline packages/core/utils/children.test.js:85]: ❤️

File: packages/core/utils/children.js [hunk 2]
```diff
@@ -49,3 +49,30 @@ export const filterChildren = (prop, value, children) => (
...
+
+/**
+ * Map an array of children into an object using the `group` prop.`
+ *
+ * @param {array} children An array of React components.
+ * @returns {object} Mapping of children by group.
+ */
+export const groupChildren = (children) => {
+  const mapping = {
+    rest: [], // Array containing ungrouped children.
```
Comment (joemcgill) [inline packages/core/utils/children.js:61]: I agree with @amyevans comment in channel that `ungrouped` would be a better name here, but overall this looks good and is really handy.

**Review by joemcgill — APPROVED** [review]
👍🏻

File: packages/core/utils/children.js (hunk 2 above)
Comment (jameswburke) [reply packages/core/utils/children.js]: Updated 👍

**Pull request merged**